test/join.jl

```diff
@@ -31,6 +31,8 @@ module TestJoin
     @test isequal(join(name, job, on = :ID, kind = :right), right)
     @test isequal(join(name, job, on = :ID, kind = :semi), semi)
     @test isequal(join(name, job, on = :ID, kind = :anti), anti)
+    @test_throws ArgumentError error(join(name, job))
```

Comment:
I mean that if you call `error` you'll always get and error, so there's nothing to test. The real test is to check that an error is thrown just when calling `join`.